test/file.test.js

```diff
@@ -258,6 +258,293 @@ describe('file', function() {
+                    'last-modified': 'Wed, 21 Oct 2015 07:28:00 GMT',
+                    'etag': ''
+                });
+            nock('http://test-status-200')
+                .get('/path/to/file.ext')
+                .reply(200, 'hello world', {
+                    'content-type': 'image/jpeg',
+                    'content-length': 11
+                });
+
+            await downloadFileConcurrently('http://test-status-200/path/to/file.ext', path.resolve('./test-transfer-file-status-200.dat'));
+            const result = await fs.readFile(path.resolve('./test-transfer-file-status-200.dat'), 'utf8');
+            assert.strictEqual(result, 'hello world');
+
+            try {
+                await fs.unlink(path.resolve('./test-transfer-file-status-200.dat'));
+            } catch (e) {
+                console.log(e);
+            }
+        });
+        it('status-200-mkdir', async function() {
+            nock('http://test-status-200')
+                .head('/path/to/file.ext')
+                .reply(200, "OK",{
+                    'content-type': 'image/jpeg',
+                    'content-length': 11,
+                    'content-disposition': 'attachment; filename="image-file-1.jpg"',
+                    'last-modified': 'Wed, 21 Oct 2015 07:28:00 GMT',
+                    'etag': ''
+                });
+            nock('http://test-status-200')
+                .get('/path/to/file.ext')
+                .reply(200, 'hello world', {
+                    'content-type': 'image/jpeg',
+                    'content-length': 11
+                });
+
+            await downloadFileConcurrently('http://test-status-200/path/to/file.ext', path.resolve('./testdir/test-transfer-file-mkdir.dat'), {
+                mkdirs: true
+            });
+            const result = await fs.readFile(path.resolve('./testdir/test-transfer-file-mkdir.dat'), 'utf8');
+            assert.strictEqual(result, 'hello world');
+
+            try{
+                await fs.unlink(path.resolve('./testdir/test-transfer-file-mkdir.dat'));
+            } catch(e){
+                // don't fail if it doesn't exist, it's only clean up
+                console.log(e);
+            }
+
+            try{
+                await fs.rmdir(path.resolve('./testdir'));
+            } catch(e){
+                // don't fail if it doesn't exist, it's only clean up
+                console.log(e);
+            }
+        });
+        it('status-200-mkdir-failure', async function() {
+            try {
+                await downloadFileConcurrently('http://test-status-200/path/to/file.ext', path.resolve('./index.js/hello.dat'), {
+                    mkdirs: true
+                });  
+                assert.fail('test is supposed to fail');
+            } catch (e) {
+                assert.ok(e.message.includes('index.js is not a directory'));
+            }
+        });
+        it.skip('status-200-truncate-retry', async function() {
```

Comment:
behavior change:
- truncated requests were retried in downloadFile and now fail in downloadFileConcurrently
   - test case reproducing it (this test)
   - downloadFile error thrown here: https://github.com/adobe/node-httptransfer/blob/master/lib/file.js#L101
   - downloadFileConcurrently error thrown here: https://github.com/adobe/node-httptransfer/blob/feature/nui-1644/lib/functions/transfer.js#L125 (does not get retried)